ports/geckolib/string_cache/regen_atom_macro.py

```diff
@@ -0,0 +1,36 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+import sys
+
+if len(sys.argv) != 2:
+    print "usage: ./%s PATH/TO/OBJDIR" % sys.argv[0]
+objdir_path = sys.argv[1]
+
+
+def line_to_atom(line):
+    result = re.match('^GK_ATOM\((.+),\s*"(.*)"\)', line)
+    return (result.group(1), result.group(2))
+
+
+def symbolify(ident):
+    return "_ZN9nsGkAtoms" + str(len(ident)) + ident + "E"
```

Comment:
Bindgen has support for this kind of things, just sayin, and while it cannot generate the macro, it does generate nicer bindings, and would avoid hardcoding the ABI mangling:

[This][bindings] is the result of:

```
./target/debug/bindgen -ignore-functions -no-type-renaming -x c++ \
                       /gecko/objdir/dist/include/nsGkAtoms.h -o nsGkAtoms.rs
```

[bindings]: https://gist.github.com/emilio/d2e1a3c8e5118310b4f61815a16c7f71